docs/usage.md

```diff
@@ -0,0 +1,323 @@
+# opentracing-cpp
+
+This file describes how clients can instrument their applications with the `opentracing-cpp`
+API. It assumes you have finished the required reading outlined [here](../README.md).
+
+### Outline
+
+The `opentracing-cpp` interface uses the [Curiously Recurring Template Pattern](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern)
+(CRTP) to define a set of compile time, polymorphic interfaces to `OpenTracing` implementations.
+
+In order to trace their systems, clients must:
+
+ - Select their `opentracing-cpp` implementation
+ - Install the selected `Tracer` in their applications in main
```

Comment:
Must this be a MUST? There are cases when we want to use different instances of the tracer in the same application.